inference-engine/src/mkldnn_plugin/nodes/mkldnn_color_convert_node.cpp

```diff
@@ -0,0 +1,666 @@
+        const auto & dims = inputDims(0);
+
+        const size_t batch_size = dims[N_DIM];
+        const size_t height = dims[H_DIM];
+        const size_t width = dims[W_DIM];
+
+        const T* y = static_cast<const T*>(input(0));
+        const T* uv = static_cast<const T*>(input(1));
+        T* dst = static_cast<T*>(output(0));
+
+        const size_t stride_y = height * width;
+        const size_t stride_uv = height * width / 2;
+
+        InferenceEngine::parallel_for2d(batch_size, height, [&](int batch, int h) {
+            typename jit_uni_converter::Params args = {
+                y + batch * stride_y + h * width,
+                uv + batch * stride_uv + (h / 2) * width,
+                dst + (batch * width * height + h * width) * 3,
+                width,
+                _colorFormat[0]     // The first byte is enough to determine the RGB or BGR format.
+            };
+
+            kernel(args);
+        });
+    }
+};
+
+}   // namespace nv12
+}   // namespace
+
+MKLDNNColorConvertNode::Converter::Converter(MKLDNNNode *node, const ColorFormat & colorFormat)
+    : _node(node)
+    , _colorFormat(colorFormat) {
+}
+
+InferenceEngine::Precision MKLDNNColorConvertNode::Converter::inputPrecision(size_t idx) const {
+    return _node->getParentEdgeAt(idx)->getMemory().GetDescWithType<BlockedMemoryDesc>()->getPrecision();
+}
+
+InferenceEngine::Precision MKLDNNColorConvertNode::Converter::outputPrecision(size_t idx) const {
+    return _node->getChildEdgeAt(idx)->getMemory().GetDescWithType<BlockedMemoryDesc>()->getPrecision();
+}
+
+const void * MKLDNNColorConvertNode::Converter::input(size_t idx) const {
+    return _node->getParentEdgeAt(idx)->getMemoryPtr()->GetPtr();
+}
+
+void * MKLDNNColorConvertNode::Converter::output(size_t idx) const {
+    return _node->getChildEdgeAt(idx)->getMemoryPtr()->GetPtr();
+}
+
+const VectorDims & MKLDNNColorConvertNode::Converter::inputDims(size_t idx) const {
+    return _node->getParentEdgesAtPort(idx)[0]->getMemory().getStaticDims();
+}
+
+MKLDNNColorConvertNode::MKLDNNColorConvertNode(const std::shared_ptr<ngraph::Node>& op,
+                                               const mkldnn::engine& eng,
+                                               MKLDNNWeightsSharing::Ptr &cache)
+    : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    std::tie(algorithm, errorMessage) = getAlgorithmFor(op);
+    if (algorithm == Algorithm::Default)
+        IE_THROW(NotImplemented) << errorMessage;
+}
+
+void MKLDNNColorConvertNode::getSupportedDescriptors() {}
+
+void MKLDNNColorConvertNode::initSupportedPrimitiveDescriptors() {
+    if (supportedPrimitiveDescriptors.empty()) {
+        switch (algorithm) {
+            case Algorithm::NV12toRGB:
+            case Algorithm::NV12toBGR: {
+                for (const auto &desc : nv12::supportedPrimitives(this))
+                    addSupportedPrimDesc(std::get<0>(desc), std::get<1>(desc), std::get<2>(desc), std::get<3>(desc));
```

Comment:
I think it would be more readable to explicitly store those parameters to local variables and then pass to addSupportedPrimDesc

Reply:
done